commons/src/main/java/org/infinispan/commons/util/SpliteratorMapper.java

```diff
@@ -44,7 +44,13 @@ public long estimateSize() {
 
    @Override
    public int characteristics() {
-      return spliterator.characteristics();
+      int characteristics = spliterator.characteristics();
+      if (mapper instanceof DistinctFunction) {
```

Comment:
I'd rather call this `InjectiveFunction`, `DistinctFunction` makes me think that I should have a counter and generate a new value every time the function is called :)